File: JavaScript/56-Merge Intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    if (!intervals) return [];
    intervals.sort(function(a,b){
        return a[0] - b[0];
    })
    let stack = [];
    for (let i=0; i<intervals.length; i++){
        if (stack.length > 0 && intervals[i][0] <= stack[stack.length-1][1]){
            stack[stack.length-1][1] = Math.max(stack[stack.length-1][1], intervals[i][1]);
        }
        else{
            stack.push(intervals[i]);
        }
    }
    return stack;
};

File: 56-Merge Intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """
        intervals.sort(key = lambda x: x[0])
        
        res = [intervals[0]]
        
        for ind, interval in enumerate(intervals):
            if interval[0] > res[-1][1]:
                res.append(interval)
            else:
                res[-1][1] = max(res[-1][1], interval[1])
                
        return res